components/create_post/create_post.jsx

```diff
@@ -365,35 +367,37 @@ export default class CreatePost extends React.Component {
             const args = {};
             args.channel_id = channelId;
             args.team_id = this.props.currentTeamId;
-            this.props.actions.executeCommand(post.message, args).then(
-                ({error}) => {
-                    this.setState({submitting: false});
-                    if (error) {
-                        if (error.sendMessage) {
-                            this.sendMessage(post);
-                        } else {
-                            this.setState({
-                                serverError: {
-                                    ...error,
-                                    submittedMessage: post.message,
-                                },
-                                message: post.message,
-                            });
-                        }
-                    }
+
+            const {error} = await this.props.actions.executeCommand(post.message, args);
+
+            if (error) {
+                if (error.sendMessage) {
+                    await this.sendMessage(post);
```

Comment:
Should we handle here the error case like in the `else` case:
 
```js
            const {error} = await this.sendMessage(post);

            if (!error) {
                this.setState({message: ''});
            }
```

Reply:
`this.state.message` will already be empty because of line 366 above, so that doesn't need to be done in case of an error. I think handling the error would be a good idea in principle, but the logic for submitting the post is already really complicated, and I don't want to risk breaking anything right now. I actually tried cleaning it up so that we could test it and such, but I had to give up after spending so much time trying to trace what happens where